src/orca/services/sevenbridges/tasks.py

```diff
@@ -46,3 +77,103 @@ def from_creds(
         factory = SevenBridgesClientFactory(api_endpoint, auth_token, client_kwargs)
         client = factory.get_client()
         return SevenBridgesTasks(client, project)
+
+    @project_required
+    def get_task(self, name: str, app_id: str) -> Optional[str]:
+        """Retrieve a task ID based on some filters.
+
+        Args:
+            name: Task name.
+            app_id: App ID.
+
+        Raises:
+            UnexpectedMatchError: If a task with the given name was found
+                but doesn't match the given app ID.
+
+        Returns:
+            The matching task ID or `None` if no matches were found.
+        """
+        project_matches = self.client.tasks.query(project=self.project)
+        name_matches = [task for task in project_matches if task.name == name]
+
+        if len(name_matches) == 0:
+            return None
+        elif len(name_matches) > 1:
+            message = f"Found many tasks ({name_matches}) with given name ({name})."
+            raise UnexpectedMatchError(message)
+        else:
+            task = name_matches[0]
+
+        if app_id not in task.app:
+            message = (
+                f"Found task ({task.id}) with given name ({name}), but its app "
+                f"({task.app}) doesn't match what's expected ({app_id})."
+            )
+            raise UnexpectedMatchError(message)
+
+        return task.id
+
+    @project_required
+    def draft_task(self, name: str, app_id: str, inputs: dict[str, Any]) -> str:
+        """Draft a task (workflow run) if need be.
+
+        This method will first query for a task with the given name.
+
+        Args:
+            name: Task name.
+            app_id: ID of the app used to draft the task.
+            inputs: Input parameters for the app.
+
+        Returns:
+            The drafted task ID.
+        """
+        # Make sure that a task doesn't already exist with that name
+        matching_task = self.get_task(name, app_id)
+        if matching_task is not None:
+            return matching_task
+
+        # Note that `create()` does not launch by default (run=False)
+        task = self.client.tasks.create(name, self.project, app_id, inputs=inputs)
+        task.id = cast(str, task.id)
+        return task.id
+
+    def launch_task(self, task_id: str) -> str:
+        """Launch a draft task (workflow run).
+
+        Args:
+            task_id: Task ID.
+
+        Returns:
+            The input task ID (passed through).
+        """
+        task = self.client.tasks.get(task_id)
```

Comment:
Does this just cause an error when a draft task doesn't exist?

Reply:
I just checked, and this is indeed the case. 

```
NotFound: Requested task does not exist.
```